server/src/main/java/io/crate/analyze/where/EqualityExtractor.java

```diff
@@ -365,15 +376,18 @@ public EqProxy add(Function compared) {
 
         static class Context {
 
-            private LinkedHashMap<ColumnIdent, Comparison> comparisons;
+            private final LinkedHashMap<ColumnIdent, Comparison> comparisons;
+            @Nullable
+            private final List<Symbol> unknownSymbols;
```

Comment:
Why not use a `Boolean` and the `seenUnknown` flag to set a value, since as far as I see we don't use the actual list of symbols, we only check if is empty or not.